query/src/main/java/org/infinispan/query/indexmanager/ClusteredSwitchingBackend.java

```diff
@@ -116,6 +116,7 @@ public void viewChanged(final ViewChangedEvent e) {
       final int viewId = e.getViewId();
       if (viewId > currentViewId) {
          if (lastSeenViewId.compareAndSet(currentViewId, viewId)) {
+            // TODO: it looks like this can be blocking - may need to offload
```

Comment:
I need input from someone who is more familiar with query if I need to move this to another thread.

Reply:
It definitely can be blocking:
* if this was the master and it has to relinquish control, it needs to flush the index and close
* if this is becoming the master, it needs to reopen the index in rw mode

Reply:
Okay, I will fix this then.